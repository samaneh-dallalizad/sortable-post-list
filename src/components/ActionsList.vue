<template>
  <li class="p-4 bg-white xl:flex" v-for="(action,index) in actions" :key="action.id">
    <div class="xl:w-2/3 lg:w-full text-slate-400 flex items-center">
      moved post {{ action.postId }} from index {{ action.from }} to index {{ action.to }}
    </div>
    <div class="xl:w-1/3 lg:w-full">
      <button
          class="btn"
          type="button"
          @click="this.$emit('timeTravel',index)"
      >
        time travel
      </button>
    </div>
  </li>
</template>

<script>
export default {
  name: "ActionsList",
  props: {
    actions: Array
  },
  emits: [
    "timeTravel"
  ],
}
</script>

<style scoped>
.btn {
  @apply py-2 px-4 rounded bg-[#25ff90] w-full
}
</style>
