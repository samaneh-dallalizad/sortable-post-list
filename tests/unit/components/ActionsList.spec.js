import {shallowMount} from '@vue/test-utils'
import ActionsList from "@/components/ActionsList";

describe('ActionsList.vue', () => {

    it('display 5 posts', () => {
        const wrapper = shallowMount(ActionsList, {
            props: {actions: [{from: 0, to: 1, postId: 1}, {from: 1, to: 2, postId: 1}, {from: 1, to: 0, postId: 3}]}
        })
        expect(wrapper.element).toMatchSnapshot()
    })
})