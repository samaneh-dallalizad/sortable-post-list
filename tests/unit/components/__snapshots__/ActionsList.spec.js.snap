// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionsList.vue display 5 posts 1`] = `
<div
  data-v-app=""
>
  
  <li
    class="p-4 bg-white xl:flex"
  >
    <div
      class="xl:w-2/3 lg:w-full text-slate-400 flex items-center"
    >
       moved post 1 from index 0 to index 1
    </div>
    <div
      class="xl:w-1/3 lg:w-full"
    >
      <button
        class="btn"
        type="button"
      >
         time travel 
      </button>
    </div>
  </li>
  <li
    class="p-4 bg-white xl:flex"
  >
    <div
      class="xl:w-2/3 lg:w-full text-slate-400 flex items-center"
    >
       moved post 1 from index 1 to index 2
    </div>
    <div
      class="xl:w-1/3 lg:w-full"
    >
      <button
        class="btn"
        type="button"
      >
         time travel 
      </button>
    </div>
  </li>
  <li
    class="p-4 bg-white xl:flex"
  >
    <div
      class="xl:w-2/3 lg:w-full text-slate-400 flex items-center"
    >
       moved post 3 from index 1 to index 0
    </div>
    <div
      class="xl:w-1/3 lg:w-full"
    >
      <button
        class="btn"
        type="button"
      >
         time travel 
      </button>
    </div>
  </li>
  
</div>
`;
